server/core/session_verification.py

```diff
@@ -1,42 +1,69 @@
-from pyraknet import c_byte
+"""
+Session verification
+"""
+
+from pyraknet.bitstream import c_byte, c_ubyte, ReadStream
+
 from plugin import Plugin, Action, Packet
 from structs import CString
 
+
 class SessionVerification(Plugin):
+    """
+    Session verification plugin
+    """
     def actions(self):
+        """
+        Returns all actions
+        """
         return [
             Action('pkt:session_info', self.verify_session, 10),
             Action('rn:user_packet', self.allow_packet, 10),
         ]
-    
+
     def packets(self):
+        """
+        Returns all packets
+        """
         return [
             SessionInfo
         ]
-    
+
     def verify_session(self, packet, address):
+        """
+        Verifies a session
+        """
         if self.server.handle_until_value('auth:check_token', True, packet.username, packet.session_key):
             self.server.connections[address]['uid'] = self.server.handle_until_value('auth:get_user_id', True, address)
         else:
             self.server.rnserver.close_connection(address)
-            
+
     def allow_packet(self, data, address):
+        """
+        Allows a packet
+        """
         packet = Packet.deserialize(ReadStream(data), self.server.packets)
-        
+
         if not getattr(packet, 'allow_without_session'):
             conn = self.server.connections.get(address)
-            if conn is None or conn.get('username') is None:
-                self.server.rnserver.close_connection(address) 
-        
+            if not conn or not conn.get('username'):
+                self.server.rnserver.close_connection(address)
+
 
 class SessionInfo(Packet):
-    allow_without_session=True
+    """
+    Session info packet
+    """
+    packet_name = 'session_info'
+    allow_without_session = True
+
     @classmethod
     def deserialize(cls, stream):
-        return cls(
-            username = stream.read(str, allocated_length=33),
-            session_key = stream.read(str, allocated_length=33),
-            unknown = stream.read(CString(allocated_length=32)),
-            unknown1 = stream.read(c_byte)
-        )
-    
+        """
+        Deserializes the packet
+        """
+        return cls(username=stream.read(str, allocated_length=33),
+                   session_key=stream.read(str, allocated_length=33),
+                   # unknown=stream.read(CString('', allocated_length=32)),
+                   unknown=stream.read(c_ubyte, allocated_length=16),  # NOTE: this seems to work, while reading a CString doesn't
```

Comment:
Could you go into more details as far as what the issue is with this? That definitely doesn't seem right.

Reply:
First option would be the best IMO, do we need `unknown` anywhere tho? If not we could just skip it couldn't we?

Reply:
It still needs to be correctly read, but yeah

Reply:
So can we skip it?

Reply:
We should get CString working anyhow, so I'd prefer just set it up with `type` for now until we can get an adjustment in pyraknet